cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,216 @@
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import lombok.Cleanup;
+import lombok.val;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+/**
+ * This command provides an administrator with the basic primitives to manipulate a DurableLog with damaged entries.
+ * The workflow of this command is as follows:
+ * 1. Checks if the Original Log is disabled (exit otherwise).
+ * 2. Reads the original damaged DurableLog and creates a backup copy of it for safety reasons.
+ * 3. Validate and buffer all the edits from admin to be done on the original DurableLog data (i.e., skip, delete,
+ * replace operations). All these changes are then written on a Repair Log (i.e., original DurableLog data + admin changes).
+ * 4. With the desired updates written in the Repair Log, the admin can replace the original DurableLog metadata by the
+ * Repair Log's one. This will make the DurableLog for the Segment Container under repair to point to the Repair Log data.
+ * 5. The backed-up data for the originally damaged DurableLog can be reused to create a new Repair Log or discarded if
+ * the Segment Container recovers as expected.
+ */
+public class DurableLogInspectCommand extends DataRecoveryCommand {
+
+    private final static Duration TIMEOUT = Duration.ofSeconds(10);
+
+    /**
+     * Creates a new instance of the DurableLogRepairCommand class.
+     *
+     * @param args The arguments for the command.
+     */
+    public DurableLogInspectCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(1);
+        int containerId = getIntArg(0);
+        val bkConfig = getCommandArgs().getState().getConfigBuilder()
+                .include(BookKeeperConfig.builder().with(BookKeeperConfig.ZK_ADDRESS, getServiceConfig().getZkURL()))
+                .build().getConfig(BookKeeperConfig::builder);
+        @Cleanup
+        val zkClient = createZKClient();
+        @Cleanup
+        DurableDataLogFactory dataLogFactory = new BookKeeperLogFactory(bkConfig, zkClient, getCommandArgs().getState().getExecutor());
+        dataLogFactory.initialize();
+
+        // Open the Original Log in read-only mode.
+        @Cleanup
+        val originalDataLog = dataLogFactory.createDebugLogWrapper(containerId);
+
+        // Check if the Original Log is disabled.
+        if (originalDataLog.fetchMetadata().isEnabled()) {
+            output("Original DurableLog is enabled. Repairs can only be done on disabled logs, exiting.");
+            return;
+        }
+
+        // Make sure that the reserved id for Backup log is free before making any further progress.
+        boolean createNewBackupLog = true;
+        if (existsBackupLog(dataLogFactory)) {
```

Comment:
Why are you keeping this code? This tool is for inspecting any log, it should not be aware of what "repair" or "backup" log is, which is specific of the repair command.

Reply:
I was under the assumption that disabled log can only be inspected as  that will make sure that we have consistent result.